Lib/test/test_subprocess.py

```diff
@@ -1485,6 +1485,50 @@ def test_exception_bad_args_0(self):
         else:
             self.fail("Expected OSError: %s" % desired_exception)
 
+    # We mock the __del__ method for Popen in the next two tests
+    # because it does cleanup based on the pid returned by fork_exec
+    # along with issuing a resource warning if it still exists. Since
+    # we don't actually spawn a process in these tests we can forego
+    # the destructor. An alternative would be to set _child_created to
+    # False before the destructor is called but there is no easy way
+    # to do that
+    @mock.patch.object(subprocess.Popen, "__del__")
```

Comment:
Good point, I didn't think about the fact that this would tie the test to an implementation detail. The subclassing solution sounds good.